packages/react-instantsearch-core/src/connectors/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,220 @@
+import algoliasearchHelper, {
+  PlainSearchParameters,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import createConnector, { ConnectedProps } from '../core/createConnector';
+import {
+  omit,
+  getObjectType,
+  removeEmptyKey,
+  removeEmptyArraysFromObject,
+} from '../core/utils';
+import {
+  refineValue,
+  getIndexId,
+  hasMultipleIndices,
+} from '../core/indexUtils';
+
+type Hit = any;
+
+export type MatchingPatterns = {
+  [attribute: string]: {
+    /**
+     * The score of the optional filter.
+     *
+     * @see https://www.algolia.com/doc/guides/managing-results/rules/merchandising-and-promoting/in-depth/optional-filters/
+     */
+    score: number;
+  };
+};
+
+interface ConfigureRelatedItemsProps {
+  /**
+   * The reference hit to extract the filters from.
+   */
+  hit: Hit;
+  /**
+   * The schema to create the optional filters.
+   * Each key represents an attribute from the hit.
+   */
+  matchingPatterns: MatchingPatterns;
+  /**
+   * Function called to transform the search parameters generated.
+   */
+  transformSearchParameters?(
+    searchParameters: SearchParameters
+  ): PlainSearchParameters;
+}
+
+function createOptionalFilter({
+  attributeName,
+  attributeValue,
+  attributeScore,
+}) {
+  return `${attributeName}:${attributeValue}<score=${attributeScore || 1}>`;
+}
+
+const defaultProps: Partial<ConfigureRelatedItemsProps> = {
+  transformSearchParameters: x => ({ ...x }),
+};
+
+function getId(): string {
+  // We store the search state of this widget in `configure`.
+  return 'configure';
```

Comment:
will this merge or override if you have a separate configure in the UI too?

Reply:
We probably need a namespace (hit's objectID)? Although OTOH it doesn't make sense to make related items multiple times?

Reply:
It merges them, as [tested here](https://github.com/algolia/react-instantsearch/blob/4ab5da03bc35e2739f9dd0ce6974d55ebb8e1840/packages/react-instantsearch-core/src/connectors/__tests__/connectConfigureRelatedItems.ts#L179-L231).

Reply:
I'm not sure it makes much sense to use multiple `ConfigureRelatedItems` widget within an `Index`, but we'll see which patterns we'll come up with when using the feature.

Reply:
but doesn't it also override the things set by `connectConfigure`, not only the things by `connectConfigureRelatedItems`? I think that's what Haroen is talking about.

Reply:
Given

```
- Index
  - Configure
  - ConfigureRelatedItems
```

If you've got `filters` in `Configure`, they'll get overridden by the ones computed in `ConfigureRelatedItems`. I don't think we should start merging them because we'll face a bunch of edge cases.

We could however display a development warning when we detect collisions and that some were overridden.

Wdyt?

Reply:
Hmm, good point, but we can avoid it with a separate key no?

Reply:
I would complexify everything just for an edge case: we'll introduce another level of nesting in the search state. I don't think we should update the structure, especially given that it's experimental. Note that we don't do this either in InstantSearch.js.

Reply:
It makes sense. It sounds good enough with dev warning 👍